File: src/main/java/com/heekng/celloct/config/oauth/dto/SessionMember.java
package com.heekng.celloct.config.oauth.dto;

import com.heekng.celloct.entity.Member;
import lombok.Getter;

import java.io.Serializable;

@Getter
public class SessionMember implements Serializable {

    private Long id;
    private String name;
    private String email;
    private String picture;

    public SessionMember(Member member) {
        this.id = member.getId();
        this.name = member.getName();
        this.email = member.getEmail();
        this.picture = member.getPicture();
    }
}

File: src/main/kotlin/com/heekng/celloct/config/oauth/dto/SessionMember.kt
package com.heekng.celloct.config.oauth.dto

import com.heekng.celloct.entity.Member

class SessionMember(
    val id: Long,
    val name: String,
    val email: String,
    val picture: String,
) : java.io.Serializable {

    companion object {
        fun fixture(member: Member): SessionMember {
            return SessionMember(
                id = member.id!!,
                name = member.name,
                email = member.email,
                picture = member.picture!!,
            )
        }
    }
}